assets/tests/components/map.test.tsx

```diff
@@ -234,6 +234,62 @@ describe("map", () => {
       screen.queryByRole("link", { name: /Go to Street View/ })
     ).not.toBeInTheDocument()
   })
+
+  test("can turn on street view and click on the map", async () => {
+    const openSpy = jest.spyOn(window, "open").mockImplementationOnce(jest.fn())
+    jest.spyOn(global, "scrollTo").mockImplementationOnce(jest.fn())
+    const mapRef: MutableRefObject<LeafletMap | null> = { current: null }
+
+    render(
+      <Map vehicles={[]} allowStreetView={true} reactLeafletRef={mapRef} />
+    )
+
+    await userEvent.click(screen.getByRole("switch", { name: "Street View" }))
+
+    await userEvent.click(mapRef.current!.getPane("mapPane")!)
+
+    expect(openSpy).toHaveBeenCalled()
+  })
+
+  test("clicking on the map with street view off doesn't open link", async () => {
+    const openSpy = jest.spyOn(window, "open").mockImplementationOnce(jest.fn())
+    jest.spyOn(global, "scrollTo").mockImplementationOnce(jest.fn())
+    const mapRef: MutableRefObject<LeafletMap | null> = { current: null }
+
+    render(
+      <Map vehicles={[]} allowStreetView={true} reactLeafletRef={mapRef} />
+    )
+
+    await userEvent.click(mapRef.current!.getPane("mapPane")!)
+
+    expect(openSpy).not.toHaveBeenCalled()
+  })
+
+  test("pressing escape leaves street view mode", async () => {
+    const mapRef: MutableRefObject<LeafletMap | null> = { current: null }
+
+    render(
+      <Map vehicles={[]} allowStreetView={true} reactLeafletRef={mapRef} />
+    )
+
+    await userEvent.click(screen.getByRole("switch", { name: "Street View" }))
+
+    await userEvent.keyboard("{Escape}")
+
+    const switchInput = screen.getByRole("switch", {
+      name: "Street View",
+    }) as HTMLInputElement
+
+    expect(switchInput.checked).toBeFalsy()
+  })
```

Comment:
suggestion: would using `checked: false` instead of `toBeFalsy` be acceptable?

https://testing-library.com/docs/queries/byrole/#checked

Reply:
Do you mean something like this?

```
    expect(
      screen.queryByRole("checked", { name: "Street View" })
    ).not.toBeInTheDocument()
```

If so I'm all for it.

Reply:
my understanding of the docs is it would be something like this?
```suggestion
    const switchInput = screen.getByRole("switch", {
      name: "Street View",
      checked: false,
    }) as HTMLInputElement

    expect(switchInput).toBeInTheDocument()
  })
```

which could compress to
```suggestion
    expect(screen.getByRole("switch", {
      name: "Street View",
      checked: false,
    })).toBeInTheDocument()
  })
```

Reply:
Ah, yes, that works. Sorry, just wasn't putting all of the pieces together quite right.